libraries/eosiolib/core/eosio/crypto_ext.hpp

```diff
@@ -48,14 +49,182 @@ namespace eosio {
          return dg;
       }
    }
-      
+
    /**
     *  @defgroup crypto Crypto
     *  @ingroup core
     *  @brief Defines API for calculating and checking hashes which
     *  require activating crypto protocol feature
     */
 
+   /**
+    * Abstracts G1 and G2 points. Ensures sizes of x and y are valid
+    *
+    *  @ingroup crypto
+    */
+   struct point_view {
+      /**
+       * Pointer to the x coordinate
+       */
+      char* x;
+
+      /**
+       * Pointer to the y coordinate
+       */
+      char* y;
+
+      /**
+       * Number of bytes in each of x and y
+       */
+      uint32_t size;
+
+      /**
+       * Construct a point given x and y
+       *
+       * @param x_ - The x coordinate, a vector of chars
+       * @param y_ - The y coordinate, a vector of chars
+       */
+      point_view(std::vector<char>& x_, std::vector<char>& y_)
+      :x(x_.data()), y(y_.data()), size(x_.size())
+      {
+         eosio::check( x_.size() == y_.size(), "x's size must be equal to y's" );
+      };
+
+      /**
+       * Construct a point given a serialized point
+       *
+       * @param p - The serialized point
+       */
+      point_view(std::vector<char>& p)
+      :x(p.data()), y(p.data() + p.size()/2), size(p.size()/2)
+      {
+      };
+
+      /**
+       *  Returns packed x and y
+       */
+      std::vector<char> packed() const {
+         std::vector<char> x_and_y( x, x + size );
+         x_and_y.insert( x_and_y.end(), y, y + size );
+         return x_and_y;
+      }
+
+      /**
+      * Unpacks a seralized point and populates current point
+      *
+      * @param packed - The source serialized point
+      */
+      void unpack(const std::vector<char>& packed) {
+         eosio::check( packed.size() == 2 * size, "size of packed point must be equal to x's size" );
+         std::memcpy(x, packed.data(), size);
+         std::memcpy(y, packed.data() + size, size);
+      }
+   };
+
+   static constexpr size_t g1_coordinate_size = 32;
+   static constexpr size_t g2_coordinate_size = 64;
+   static constexpr size_t blake2f_result_size = 64;
+
+   /**
+    * Abstracts G1 point. Ensures sizes of x and y are 32
+    *
+    *  @ingroup crypto
+    */
+   struct g1_view : public point_view {
+      /**
+       * Construct a G1 point given x and y
+       *
+       * @param x_ - The x coordinate, a vector of chars
+       * @param y_ - The y coordinate, a vector of chars
+       */
+      g1_view(std::vector<char>& x_, std::vector<char>& y_)
```

Comment:
Thanks for your suggestion here and in the associated issue. `std::span` is in C++20 while we still need to support C++17 at the moment.

Reply:
Oh sorry I didn't realize support for C++17 is still required.

Ok I guess this would be proper for separate issue/PR, but how about creating a new span like type for C++17 (internal namespace) and exposing only 'bytes view' like span alias to cdt? When support for C++17 is dropped in the future only this internal span implementation has to be removed but the alias would stay intact without breaking api.

Just for reference, I made one simple type imitating span for such cases: https://github.com/ZeroPass/antelope.ck/blob/1bf203db28d053e9d5b171fb92f27c44b94acdc4/include/eosiock/span.hpp